src/Cli/dotnet/Commands/Test/MTP/Terminal/TestProgressState.cs

```diff
@@ -122,8 +126,8 @@ public void ReportFailedTest(string testNodeUid, string instanceId)
 
     public void DiscoverTest(string? displayName, string? uid)
     {
-        PassedTests++;
-        DiscoveredTests.Add(new(displayName, uid));
+        DiscoveredTests++;
```

Comment:
Unrelated to your changes. But do we guarantee this is not called concurrently from multiple threads? This is not only for discovery, but for reporting test results as well.